pkg/config/templates/rancherd-10-harvester.yaml

```diff
@@ -202,6 +202,10 @@ resources:
         enabled: true
       kube-vip-cloud-provider:
         enabled: true
+      promote:
```

Comment:
My initial thought is to communicate that, within the `ManagedChart`, these settings are only used by the promotion controller, instead of a general mechanism to override RKE2, per github.com/harvester/harvester/pull/7156. My current preference leans towards keeping it as `promote`. LMKWYT.

Re: CNI, can we work on it as a separate issue?

Reply:
Sure, customized CNI will be in another issue & PR, which needs some investigation.

`promotion` is also good.

Reply:
Custom CNI enhancement request added at #7197.